ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -143,6 +143,36 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
 		}
 		// TODO (kevin85421): Don't return here; we still need to move some of the code below into this block.
+	case rayv1.JobDeploymentStatusSuspending:
+		// The `suspend` operation should be atomic. In other words, if users set the `suspend` flag to true and then immediately
+		// set it back to false, either all of the RayJob's associated resources should be cleaned up, or no resources should be
+		// cleaned up at all. To keep the atomicity, if a RayJob is in the `Suspending` status, we should delete all of its
+		// associated resources and then transition the status to `Suspended` no matter the value of the `suspend` flag.
+
+		// TODO (kevin85421): Currently, Ray doesn't have a best practice to stop a Ray job gracefully. At this moment,
+		// KubeRay doesn't stop the Ray job before suspending the RayJob. If users want to stop the Ray job by SIGTERM,
+		// users need to set the Pod's preStop hook by themselves.
```

Comment:
do we document this anywhere other than here in code?

Reply:
Currently not. I will streamline the end-to-end UX of RayJob with some Kubernetes maintainers from Google this quarter. I will document important information when we figure out the best practice.